app/utils/AffinityGroupUtils.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import play.api.mvc.Request
+
+object AffinityGroupUtils {
+
+  def isAgent[A](request: Request[A]): Boolean = {
```

Comment:
Rather than doing this approach which assumes a session value has been set before they enter out service. We should be using `IdentifierAction` in the controller and adding the AffinityGroup to `IdentifierRequest`. As we cannot guarantee the source of the request due to load balancers this will fail if a request is sent to another microservice instance from NGINX.
Can we refactor this approach and not use a `Session`.
We will be able to identify what type the user is through the implicit request.
i.e:
```scala
case class IdentifierRequest[A] (request: Request[A], identifier: String, affinityGroup: AffinityGroup) 
  extends WrappedRequest[A](request)
```

Reply:
Done